moonraker/components/mqtt.py

```diff
@@ -247,6 +247,10 @@ async def component_init(self) -> None:
                 logging.info("Unable to connect to MQTT broker, "
                              f"retries remaining: {retries}")
                 await asyncio.sleep(2.)
+            except socket.gaierror:
+                logging.info("Unable to resolve MQTT broker hostname, "
+                             f"retries remaining: {retries}")
+                await asyncio.sleep(2.)
```

Comment:
I think we can handle this in a single except statement with something like the following:
```python
for _ in range(retries):
    try:
        self.client.connect(self.address, self.port)
    except (ConnectionRefusedError, socket.gaierror) as e:
        logging.info(f"MQTT connection error, {e}, "
                     "retries remaining: {retries}")
        await asyncio.sleep(2.)
```

Reply:
Done!

FYI, this is output right now:

```text
MQTT connection error, [Errno -3] Temporary failure in name resolution, retries remaining: 5
```

Reply:
That was fast!.  I think that is fine.  If necessary we can refactor the error message later.